homework/index.js

```diff
@@ -29,19 +14,113 @@
     });
     return elem;
   }
+  const fetchJSON = url =>
+    fetch(url)
+      .then(res => {
+        if (res.ok) {
+          return res;
+        }
+        throw new Error('Network request failed');
+      })
+      .then(res => res.json());
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  const showError = err => {
+    const root = document.getElementById('root');
+
+    createAndAppend('div', root, {
+      text: 'HYF Repositories',
+      class: 'HYF_Repositories',
+      id: 'HYF_Repositories',
+    });
+    createAndAppend('div', root, {
+      text: `${err.message}`,
+      class: 'alert-error',
+    });
+  };
+
+  /* cSpell:disable */
+  function makeApp(jsonFile) {
+    jsonFile.sort((a, b) => a.name.localeCompare(b.name));
+    const root = document.getElementById('root');
+    const hyfRepositories = createAndAppend('div', root, {
+      class: 'HYF_Repositories',
+      id: 'HYF_Repositories',
+    });
+    createAndAppend('h1', hyfRepositories, { text: 'HYF Repositories' });
+    const repositories = createAndAppend('select', hyfRepositories, { id: 'repositories' });
+    const informationBox = createAndAppend('div', root, {
+      class: 'information_box',
+      id: 'information_box',
+    });
+    const repositoryInformation = createAndAppend('ul', informationBox, {
+      id: 'repository_information',
+    });
+    const contributions = createAndAppend('ul', informationBox, {
+      text: 'Contributors',
+      id: 'contributions',
+      class: 'right_ul',
+    });
+    const renderContributors = contributor => {
+      contributions.innerHTML = 'Contributors';
```

Comment:
Since `contributions` is the `ul` list adding the word Contributors will resolve in a invalid html list, the text should be above the list.